File: src/platform/ffpython.py
# -*- coding: utf-8 -*-
"""platform/ffpython.py — 定位 FontForge 的 ffpython.exe。"""
from __future__ import annotations
import shutil
from pathlib import Path

CANDIDATES = [
    r"C:\Program Files\FontForgeBuilds\bin\ffpython.exe",
    r"C:\Program Files (x86)\FontForgeBuilds\bin\ffpython.exe",
]


def find_ffpython() -> str | None:
    """返回可用的 ffpython 路径，找不到返回 None。"""
    for cand in CANDIDATES:
        if Path(cand).exists():
            return cand
    found = shutil.which("ffpython")
    return found

File: src/main/ffpython.ts
import { spawnSync } from 'child_process'
import { existsSync } from 'fs'
import { join } from 'path'

export function findFfpython(): string | null {
  const candidates = [
    join('C:', 'Program Files', 'FontForgeBuilds', 'bin', 'ffpython.exe'),
    join('C:', 'Program Files (x86)', 'FontForgeBuilds', 'bin', 'ffpython.exe')
  ]
  for (const candidate of candidates) {
    if (existsSync(candidate)) return candidate
  }
  const result = spawnSync('where', ['ffpython'], { encoding: 'utf-8', timeout: 5000 })
  if (result.status === 0 && result.stdout.trim()) {
    return result.stdout.trim().split('\n')[0].trim()
  }
  return null
}
